### ISSUE TITLE
Unable to eth.sendTransaction on 1.10.5 with client in light sync mode.

### ISSUE THREAD

**Issue by JoeGruffins** [ethereum/go-ethereum#23214]
#### System information

Geth version: 1.10.5-stable
OS & Version: linux 5.12.13-arch1-2

#### Expected behaviour

In console with an unlocked account sending the command:
```
eth.sendTransaction({from:eth.accounts[0],to:"0x18d65fb8d60c1199bb1ad381be47aa692b482605",value:1000,gasPrice:82000000000})
```
sends eth to the account.

#### Actual behaviour

Error.

#### Steps to reproduce the behaviour

Unlock account with node in light mode. Attempt `eth.sendTransaction`

#### Backtrace

````
eth.sendTransaction({from:eth.accounts[0],to:"0x18d65fb8d60c1199bb1ad381be47aa692b482605",value:1000,gasPrice:82000000000})
Error: block not found
        at web3.js:6357:37(47)
        at web3.js:5091:62(37)
        at <eval>:1:20(15)
````

#### Misc

Performing this exact same operation in 1.10.4 worked. I am using a private network and so it could be something in my setup. I do not have any funds to try on testnet or mainnet. It does work with snap sync nodes on the same network.

**Comment by karalabe** [comment]
Thanks for the report. Can confirm on testnet.

**Comment by rjl493456442** [comment]
Thanks, fixed in the #23215!
